examples/clients/opengl-sdl/OpenGLSample.cpp

```diff
@@ -0,0 +1,290 @@
+                break;
+            case SDL_KEYDOWN:
+                if (SDL_SCANCODE_ESCAPE == e.key.keysym.scancode) {
+                    // Handle pressing ESC
+                    quit = true;
+                }
+                break;
+            }
+            if (e.type == SDL_QUIT) {
+                quit = true;
+            }
+        }
+
+        // Update OSVR
+        ctx.update();
+
+        // Render
+        bool valid = render(display);
+
+        // Swap buffers if we had all the data to render.
+        if (valid) {
+            SDL_GL_SwapWindow(window.get());
+        }
+    }
+
+    return 0;
+}
+
+/// @brief A simple dummy "draw" function - note that drawing occurs in "room
+/// space" by default. (that is, in this example, the modelview matrix when this
+/// function is called is initialized such that it transforms from world space
+/// to view space)
+void renderScene() { draw_cube(1.0); }
+
+/// @brief The "wrapper" for rendering to a device described by OSVR.
+///
+/// This function will set up viewport, initialize view and projection matrices
+/// to current values, then call `renderScene()` as needed (e.g. once for each
+/// eye, for a simple HMD.)
+///
+/// @return false if we didn't render (typically because we just started up and
+/// don't have pose data for head/eyes yet)
+bool render(OSVR_DisplayConfig disp) {
+    /// For each viewer...
+    OSVR_ViewerCount viewers;
+    osvrClientGetNumViewers(disp, &viewers);
+    for (OSVR_ViewerCount viewer = 0; viewer < viewers; ++viewer) {
+
+        /// For each eye of the given viewer...
+        OSVR_EyeCount eyes;
+        osvrClientGetNumEyesForViewer(disp, viewer, &eyes);
+        for (OSVR_EyeCount eye = 0; eye < eyes; ++eye) {
+
+            /// Try retrieving the view matrix (based on eye pose) from OSVR
+            double viewMat[OSVR_MATRIX_SIZE];
+            auto gotView = osvrClientGetViewerEyeViewMatrixd(
+                disp, viewer, eye,
+                OSVR_MATRIX_COLMAJOR | OSVR_MATRIX_COLVECTORS, viewMat);
+
+            if (gotView != OSVR_RETURN_SUCCESS) {
+                std::cout << "Waiting for view pose..." << std::endl;
+                return false;
+            }
+            static bool announcedPose = false;
+            if (!announcedPose) {
+                std::cout << "Got view pose, rendering will commence!"
+                          << std::endl;
+                announcedPose = true;
+            }
+
+            /// Initialize the ModelView transform with the view matrix we
+            /// received
+            glMatrixMode(GL_MODELVIEW);
+            glLoadIdentity();
+            glMultMatrixd(viewMat);
+
+            /// For each display surface seen by the given eye of the given
+            /// viewer...
+            OSVR_SurfaceCount surfaces;
+            osvrClientGetNumSurfacesForViewerEye(disp, viewer, eye, &surfaces);
+            for (OSVR_SurfaceCount surface = 0; surface < surfaces; ++surface) {
```

Comment:
In RenderManager, it is the problem of RenderManager because it set up the displays and such.  So, as I'm switching from render-to-screen to render-to-texture this is changing under the covers.  This is why we need all these callbacks to the app: start display, start eye, end eye, end display.  If the app needs to own this, it is hideously complicated and special-case and you get code drift.  That's why I'm trying to pull it into OSVR.  My target is an app that can write to different surfaces just based on the config file.  I think I'm getting there, because the example apps should not be changing as I swap out render-texture underneath them...